src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -230,89 +174,76 @@ int ACA_Dataplane_OVS::update_port_state_workitem(const PortState current_PortSt
+
+      break;
+
+    case OperationType::UPDATE:
+      // only delete scenario is supported now
+      // VM was created with port specified, then delete the VM
+      // ACA will receive update with no device_id and device_owner
+      if (!current_PortConfiguration.device_id().empty() ||
+          !current_PortConfiguration.device_owner().empty()) {
+        throw std::invalid_argument("Port Operation: update but device_id or device_owner not empty");
       }
-
-    } catch (const std::invalid_argument &e) {
-      ACA_LOG_ERROR("Invalid argument exception caught while parsing port configuration, message: %s.\n",
-                    e.what());
-      overall_rc = -EINVAL;
-    } catch (const std::exception &e) {
-      ACA_LOG_ERROR("Exception caught while parsing port configuration, message: %s.\n",
-                    e.what());
-      overall_rc = -EFAULT;
-    } catch (...) {
-      ACA_LOG_CRIT("%s", "Unknown exception caught while parsing port configuration.\n");
-      overall_rc = -EFAULT;
+      [[fallthrough]];
+    case OperationType::DELETE:
+      // another delete scenario is supported here
+      // VM was created with without port specified, then delete the VM
+      // ACA will receive port operation delete
+      ACA_LOG_DEBUG("Port Operation: %s: port_id: %s, project_id:%s, vpc_id:%s, network_type:%d, "
+                    "generated_port_name: %s, tunnel_id: %d\n",
+                    aca_get_operation_string(current_PortState.operation_type()),
+                    current_PortConfiguration.id().c_str(),
+                    current_PortConfiguration.project_id().c_str(),
+                    current_PortConfiguration.vpc_id().c_str(), found_network_type,
+                    generated_port_name.c_str(), found_tunnel_id);
+
+      overall_rc = ACA_OVS_L2_Programmer::get_instance().delete_port(
+              current_PortConfiguration.vpc_id(), generated_port_name,
+              found_tunnel_id, culminative_dataplane_programming_time);
+
+      break;
+
+    default:
+      ACA_LOG_ERROR("Invalid port state operation type %d\n",
+                    current_PortState.operation_type());
+      overall_rc = -EXIT_FAILURE;
+      break;
     }
 
-    break;
-
-  default:
-    ACA_LOG_ERROR("Invalid port state operation type %d\n",
-                  current_PortState.operation_type());
-    overall_rc = -EXIT_FAILURE;
-    break;
+  } catch (const std::invalid_argument &e) {
+    ACA_LOG_ERROR("Invalid argument exception caught while parsing port configuration, message: %s.\n",
+                  e.what());
+    overall_rc = -EINVAL;
+  } catch (const std::exception &e) {
+    ACA_LOG_ERROR("Exception caught while parsing port configuration, message: %s.\n",
+                  e.what());
+    overall_rc = -EFAULT;
+  } catch (...) {
+    ACA_LOG_CRIT("%s", "Unknown exception caught while parsing port configuration.\n");
+    overall_rc = -EFAULT;
   }
 
+Done:
```

Comment:
Done?

Reply:
yes, I don't like using goto, but this is a cleaner way I can think of to handle the error condition in line 162. Let me know if you have a better suggestion.

Reply:
I think you might mean "ready to exit" :-)

Reply:
renamed to "EXIT".